mongo/src/test/java/com/redhat/lightblue/mongo/crud/MongoCRUDControllerTest.java

```diff
@@ -90,6 +102,250 @@ public DB get(MongoDataStore store) {
+
+
+        ObjectField o = new ObjectField("field2");
+        o.getFields().put(new SimpleField("x", StringType.TYPE));
+
+        SimpleArrayElement saSub = new SimpleArrayElement(StringType.TYPE);
+        ArrayField afSub = new ArrayField("subArrayField", saSub);
+        o.getFields().put(afSub);
+
+        ObjectArrayElement oaObject = new ObjectArrayElement();
+        oaObject.getFields().put(new SimpleField("x", StringType.TYPE));
+
+        SimpleArrayElement saSubSub = new SimpleArrayElement(StringType.TYPE);
+        ArrayField afSubSub = new ArrayField("arraySubObj", saSubSub);
+        oaObject.getFields().put(afSubSub);
+
+        ArrayField afObject = new ArrayField("arrayObj", oaObject);
+        e.getFields().put(afObject);
+
+        e.getFields().put(o);
+
+
+        SimpleArrayElement sa = new SimpleArrayElement(StringType.TYPE);
+        ArrayField af = new ArrayField("arrayField", sa);
+        e.getFields().put(af);
+
+        e.getEntityInfo().setDefaultVersion("1.0.0");
+        e.getEntitySchema().getAccess().getInsert().setRoles("anyone");
+        e.getEntitySchema().getAccess().getFind().setRoles("anyone");
+        return e;
+    }
+
+    private EntityMetadata addCIIndexes(EntityMetadata e) {
+        Index index1 = new Index();
+        index1.setName("testIndex1");
+        index1.setUnique(true);
+        List<IndexSortKey> indexFields1 = new ArrayList<>();
+        indexFields1.add(new IndexSortKey(new Path("field1"), true));
+        indexFields1.add(new IndexSortKey(new Path("field3"), true, true));
+        indexFields1.add(new IndexSortKey(new Path("arrayField.*"), true, true));
+        indexFields1.add(new IndexSortKey(new Path("field2.x"), true, true));
+        index1.setFields(indexFields1);
+
+        Index index2 = new Index();
+        index2.setName("testIndex2");
+        index2.setUnique(true);
+        List<IndexSortKey> indexFields2 = new ArrayList<>();
+        indexFields2.add(new IndexSortKey(new Path("arrayObj.*.x"), true, true));
+        index2.setFields(indexFields2);
+
+        Index index3 = new Index();
+        index3.setName("testIndex3");
+        index3.setUnique(true);
+        List<IndexSortKey> indexFields3 = new ArrayList<>();
+        indexFields3.add(new IndexSortKey(new Path("arrayObj.*.arraySubObj.*"), true, true));
+        index3.setFields(indexFields3);
+
+        Index index4 = new Index();
+        index4.setName("testIndex4");
+        index4.setUnique(true);
+        List<IndexSortKey> indexFields4 = new ArrayList<>();
+        indexFields4.add(new IndexSortKey(new Path("field2.subArrayField.*"), true, true));
+        index4.setFields(indexFields4);
+
+
+        List<Index> indexes = new ArrayList<>();
+        indexes.add(index1);
+        indexes.add(index2);
+        indexes.add(index3);
+        indexes.add(index4);
+        e.getEntityInfo().getIndexes().setIndexes(indexes);
+        return e;
+    }
+
```

Comment:
Add a test that verifies an already existing index is not recreated upon metadata update